src/systems/joint_trajectory_controller/JointTrajectoryController.cc

```diff
@@ -0,0 +1,792 @@
+  /// \param[in] _ecm Ignition Entity Component Manager
+  /// \param[in] _enabledJoints List of all joints that should be enabled based on SDF configuration
+  public: void ConfigureJoint(const Entity &_entity,
+                              const std::shared_ptr<const sdf::Element> &_sdf,
+                              const ignition::gazebo::EntityComponentManager &_ecm,
+                              const std::vector<std::string> &_enabledJoints);
+
+  /// \brief Reset internals of the plugin, without affecting already created components
+  public: void Reset();
+
+  /// \brief Ignition communication node
+  public: transport::Node node;
+
+  /// \brief Publisher of the progress for currently followed trajectory
+  public: transport::Node::Publisher progressPub;
+
+  /// \brief Map of actuated joints, where first is the name of the joint
+  public: std::map<std::string, ActuatedJoint> actuatedJoints;
+
+  /// \brief Mutex projecting trajectory
+  public: std::mutex trajectoryMutex;
+
+  /// \brief Information about trajectory that should be followed
+  public: Trajectory trajectory;
+
+  /// \brief Flag that determines whether to use message header timestamp as the trajectory start,
+  /// where simulation time at the beginning of execution is used otherwise
+  public: bool useHeaderStartTime;
+
+  /// \brief Flag that determines if all components required for control are already setup
+  public: bool componentSetupFinished;
+};
+
+////////////////////////
+/// Helper Functions ///
+////////////////////////
+
+/// \brief Parse separated parameters from a string
+/// \param[in] _input A string that has multiple separated parameters, e.g. with whitespace
+/// \return Vector of all T-typed parameters contained in the input
+template <typename T>
+std::vector<T> ParseVectorParam(const std::string &_input)
+{
+  std::istringstream iss(_input);
+  std::vector<T> output;
+  T value;
+
+  while (iss >> value)
+  {
+    output.push_back(value);
+  }
+  return output;
+}
+
+/// \brief Return value from `_vec` at `_index`, or `_default` if `_vec` is not large enough
+/// \param[in] _vec Vector that contains desired value of type T
+/// \param[in] _index Index at which the value is stored
+/// \param[in] _default Default value of type T
+/// \return Value from `_vec` at `_index`, or `_default` if `_vec` is not large enough
+template <typename T>
+T GetNthOrDefault(const std::vector<T> &_vec,
+                  const size_t &_index,
+                  const T &_default)
+{
+  if (_index < _vec.size())
+  {
+    return _vec[_index];
+  }
+  else
+  {
+    return _default;
+  }
+}
+
+/////////////////////////////////
+/// JointTrajectoryController ///
+/////////////////////////////////
+
+JointTrajectoryController::JointTrajectoryController()
+    : dataPtr(std::make_unique<JointTrajectoryControllerPrivate>())
+{
+}
+
+void JointTrajectoryController::Configure(const Entity &_entity,
```

Comment:
Nit: add 50 `/` before each function:

```suggestion
//////////////////////////////////////////////////
void JointTrajectoryController::Configure(const Entity &_entity,
```

Reply:
Added everywhere as part of https://github.com/ignitionrobotics/ign-gazebo/pull/473/commits/ff6c82eacc6c15e74d8b7a568f4dcdf3577c78a5